Level manager

TalausLabs opened the pull request.

Gracie-Gray committed:
Implemented CollisionComponent and Platform. Edited Entity to include name, position, etc. Attempting to add drawn rectangles as platforms.

Gracie-Gray committed:
Collision Detection seems to work. Platform rendering still invisible. Possibly due to movement rendering in render().

Gracie-Gray committed:
commiting and pushing from VS Code, netbeans not authenticated to push??

Gracie-Gray committed:
Added utilities CollisionDetection. Platform rendering still invisible. Not sure why.

Gracie-Gray committed:
Pushing from VScode

Gracie-Gray committed:
PLatform appera. Hitboxes move. Collision Detection works.Added plasma ball pngs.

Gracie-Gray committed:
Comitting by VScode again

TalausLabs committed:
Implemented collision detection & jump feature

Updated loadimage method in sprite class to allow for scaling the image
updated player constructor to scale the player sprite to the size of the hitbox so it matches.
Reworked playermovementstrategy to allow jump feature & gravity
Added collision detection to playermovementstrategy

TalausLabs committed:
removed hitbox offset test value

removed hitbox offset test value

1leo1tro committed:
Plasma ball

Still have to debug/add sprite

1leo1tro committed:
Fixed Bugs

Fixed weird oval around player and ball disappears upon collision

1leo1tro committed:
Plasma ball animation

TalausLabs committed:
fixed player white space & added boundaries on edge of map

TalausLabs committed:
successful level system & manager

TalausLabs committed:
temp

TalausLabs committed:
temp2

TalausLabs committed:
implemented jumppad changes to this branch

Aeyu17 committed:
Merge main into level-manager

Aeyu17 committed:
Fix more merge issues

Aeyu17 reviewed (dismissed):
Make the level manager able to switch between different screens rather than clearing all entities and loading a new set of items; this will make it significantly easier to design multiple levels

TalausLabs committed:
level manager update

designed the level layout, for level 1 and added tests to switch between levels

Aeyu17 reviewed (approved):
Looks good to me
